DOC Add info for links to docs and images for modules (#780)
silverstripe/developer-docs: en/10_Contributing/02_Documentation.md

### Links to documentation

> When updating documentation for [`silverstripe/developer-docs`](https://github.com/silverstripe/developer-docs/), which is the content repository for [`docs.silverstripe.org`](https://docs.silverstripe.org), links to other pages in the documentation should use the URL slug, not the file name.

Reviewer: ```suggestion
When updating documentation for [`silverstripe/developer-docs`](https://github.com/silverstripe/developer-docs/), which is the content repository for [`docs.silverstripe.org`](https://docs.silverstripe.org), links to other pages in the documentation should use the URL slug (e.g. `/contributing/documentation/`), not the file name.
```

Author: Updated

> When updating docs that are in a modules `docs/en` folder, and will be shown in the "Optional features" section of `docs.silverstripe.org`, and are also intended to be readible on GitHub, then links to other pages should use a relative path to the file name, not the URL slug.

Reviewer: ```suggestion
When updating docs that are in a module's `docs/en/` folder, and will be shown in the "Optional features" section of `docs.silverstripe.org`, and are also intended to be readible on GitHub, then links to other pages within that module should use a relative path to the file name, not the URL slug. Links to documentation outside that module should use a full URL without the version number (e.g. `https://docs.silverstripe.org/en/contributing/documentation/`). API doc links should still be used in the format described below.
```

Reviewer: Also, put this note _below_ the bits about "Relative documentation links" and "Absolute documentation links" below which are specific to docs in `silverstripe/developer-docs`.

Author: Updated

When updating docs that are in a module's `docs/en/` folder, and will be shown in the "Optional features" section of `docs.silverstripe.org`, and are also intended to be readible on GitHub, then links to other pages within that module should use a relative path to the file name, not the URL slug. Links to documentation outside that module should use a full URL without the version number (e.g. `https://docs.silverstripe.org/en/contributing/documentation/`). API doc links should still be used in the format described below.

> Absolute documentation links are preferred for pages which aren't in the same directory or one level up. The language and version number are ommitted. For example `[managing lists](/developer_guides/model/lists/)` becomes [managing lists](/developer_guides/model/lists/).

Reviewer: This should go above the "updating docs that are in a module's `docs/en/` folder" bit (as per https://github.com/silverstripe/developer-docs/pull/780#discussion_r2214583039) since it's talking specifically about how to format links in the result developer-docs docs.

Reviewer 2: readible is not a word -> readable or legible

Author: Updated

### Images

> Add images, such as screenshots, to an `_images` directory which should usually be located at `docs/en/_images`. Use relative links these files and use the markdown syntax `![Some alt text](_images/some-file.jpg)`. You must include alt text. Adding images this way will work correctly on both markdown added to `silverstripe/developer-docs`, and also to `docs/en` folder for a module that dispays on both `docs.silverstripe.org` and GitHub.

Reviewer: ```suggestion
Add images, such as screenshots, to an `_images/` directory which should usually be located at `docs/en/_images/`. Use relative links these files and use the markdown syntax `![Some alt text](_images/some-file.jpg)`. You must include alt text. Adding images this way will work correctly on both markdown added to `silverstripe/developer-docs`, and also to `docs/en/` folder for a module that dispays on both `docs.silverstripe.org` and GitHub.
```

Author: Updated
